src/main/java/org/radarbase/kafka/connect/transforms/KeyValueTransform.java

```diff
@@ -105,7 +105,13 @@ private R applySchemaless(R record) {
   }
 
   private long convertTimestamp(Object time){
-    return (long) (Double.parseDouble(time.toString()) * 1000);
+      try {
+        return (long)(new Double(time.toString()) * 1000);
```

Comment:
I was thinking more along the lines of

```java
if (time instanceof Double) {
  return (long) Math.round((Double) time * 1000.0);
} else {
  return time; // if it’s not a double, the conversion may not be correct. Skip conversion.
}
```

Reply:
Oh okay I see. Thanks. Can I cast the time to `long` in the else statement since we are returning `long`?

Reply:
Better to return Object then

Reply:
Okay sure thanks